### PR TITLE
Move premerge to linux agents + annotate nunit test failures

### PR DESCRIPTION
**Pull request by paulbalaji**
- move premerge from windows to linux agents
- dockerise build/test
- add plugin to annotate nunit test failures

### CODE REVIEW

File: ci/build-test.sh [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
+setupPremerge
 
-echo "--- Test DocsLinter.csproj :link:"
-dotnet test --logger:"nunit;LogFilePath=$(pwd)/logs/docslinter-test-results.xml" "tools/DocsLinter/DocsLinter.csproj"
+mkdir -p ./logs
 
-echo "--- Test ReleaseTool.csproj :fork:"
-dotnet test --logger:"nunit;LogFilePath=$(pwd)/logs/releasetool-test-results.xml" "tools/ReleaseTool.Tests/ReleaseTool.Tests.csproj"
+echo "--- Running premerge :running:"
+docker run \
```
Comment (zeroZshadow) [inline ci/build-test.sh]: This should probable have `--rm`
https://docs.docker.com/engine/reference/run/#clean-up---rm

File: ci/common.sh [hunk 2]
```diff
@@ -30,6 +30,15 @@ function setupReleaseTool() {
         .
 }
 
+function setupPremerge() {
+    echo "--- Setting up premerge :gear:"
+
+    docker build \
+        --tag local:shared-ci-premerge \
```
Comment (zeroZshadow) [inline ci/common.sh]:
```suggestion
        --tag shared-ci-premerge \
```

File: ci/build-test.sh (hunk 1 above)
Comment (zeroZshadow) [inline ci/build-test.sh]:
```suggestion
docker run --rm \
```

File: ci/build-test.sh [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
+    -v "$(pwd)"/logs:/var/logs \
+    local:shared-ci-premerge
```
Comment (zeroZshadow) [inline ci/build-test.sh]:
```suggestion
    shared-ci-premerge
```

File: ci/build-test.sh [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
+### automation, but not much more than that.
+
 set -e -u -o pipefail
 
 if [[ -n "${DEBUG-}" ]]; then
   set -x
 fi
 
+if [[ -z "$BUILDKITE" ]]; then
+  echo "This script is only to be run on Improbable CI."
```
Comment (jamiebrynes7) [inline ci/build-test.sh]: Why?

File: ci/docker/premerge.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,14 @@
+FROM microsoft/dotnet:2.2-sdk as build
+WORKDIR /app
+COPY ./tools ./tools/
+COPY ./ci/docker/build-test-entrypoint.sh ./build-test.sh
```
Comment (jamiebrynes7) [inline ci/docker/premerge.Dockerfile]: Nit: Rename `build-test-entrypoint.sh` to just `entrypoint.sh` and `./build-test.sh` to `./entrypoint.sh`

Makes it clearer as to what is going on

**Review by zeroZshadow — APPROVED** [review]
LGTM % suggestions

File: ci/common.sh [hunk 6]
```diff
@@ -30,6 +30,15 @@ function setupReleaseTool() {
         .
 }
 
+function setupPremerge() {
```
Comment (jamiebrynes7) [inline ci/common.sh]: Tbh lets just inline this to where its used

**Review by jamiebrynes7 — APPROVED** [review]

**Review by jessicafalk — APPROVED** [review]

**Pull request merged**